<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1064</width>
    <height>636</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>540</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Heiti SC</family>
     </font>
    </property>
    <property name="text">
     <string>file name:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>570</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Heiti SC</family>
     </font>
    </property>
    <property name="text">
     <string>number of verticles:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>600</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Heiti SC</family>
     </font>
    </property>
    <property name="text">
     <string>number of edges:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="open_file">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>30</y>
      <width>121</width>
      <height>32</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Heiti SC</family>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(94, 178, 255);
border-radius:3px;</string>
    </property>
    <property name="text">
     <string>Open file</string>
    </property>
   </widget>
   <widget class="QPushButton" name="to_JPG">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>550</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Heiti SC</family>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 189, 101);
border-radius:3px;</string>
    </property>
    <property name="text">
     <string>PHOTO</string>
    </property>
   </widget>
   <widget class="QPushButton" name="to_GIF">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>550</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Heiti SC</family>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 189, 101);
border-radius:3px;</string>
    </property>
    <property name="text">
     <string>GIF</string>
    </property>
   </widget>
   <widget class="QLabel" name="file_name">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>540</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Heiti SC</family>
     </font>
    </property>
    <property name="text">
     <string>...</string>
    </property>
   </widget>
   <widget class="QLabel" name="num_of_verticles">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>570</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Heiti SC</family>
     </font>
    </property>
    <property name="text">
     <string>...</string>
    </property>
   </widget>
   <widget class="QLabel" name="num_of_edges">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>600</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Heiti SC</family>
     </font>
    </property>
    <property name="text">
     <string>...</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>70</y>
      <width>167</width>
      <height>436</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_7">
     <item row="0" column="0">
      <widget class="QLabel" name="label_4">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>translate</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0" rowspan="3">
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0" rowspan="3">
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QLabel" name="label_5">
             <property name="font">
              <font>
               <family>Heiti SC</family>
              </font>
             </property>
             <property name="text">
              <string>X:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_6">
             <property name="font">
              <font>
               <family>Heiti SC</family>
              </font>
             </property>
             <property name="text">
              <string>Y:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_7">
             <property name="font">
              <font>
               <family>Heiti SC</family>
              </font>
             </property>
             <property name="text">
              <string>Z:</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="translate_left_x">
           <property name="text">
            <string>&lt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="translate_right_x">
           <property name="text">
            <string>&gt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="translate_left_y">
           <property name="text">
            <string>&lt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="translate_right_y">
           <property name="text">
            <string>&gt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="translate_left_z">
           <property name="text">
            <string>&lt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QPushButton" name="translate_right_z">
           <property name="text">
            <string>&gt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="tr_x">
         <property name="font">
          <font>
           <family>Heiti SC</family>
          </font>
         </property>
         <property name="text">
          <string>0.00</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="tr_y">
         <property name="font">
          <font>
           <family>Heiti SC</family>
          </font>
         </property>
         <property name="text">
          <string>0.00</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="tr_z">
         <property name="font">
          <font>
           <family>Heiti SC</family>
          </font>
         </property>
         <property name="text">
          <string>0.00</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_17">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>rotate</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0" rowspan="3">
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0" rowspan="3">
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QLabel" name="label_11">
             <property name="font">
              <font>
               <family>Heiti SC</family>
              </font>
             </property>
             <property name="text">
              <string>X:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_12">
             <property name="font">
              <font>
               <family>Heiti SC</family>
              </font>
             </property>
             <property name="text">
              <string>Y:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_13">
             <property name="font">
              <font>
               <family>Heiti SC</family>
              </font>
             </property>
             <property name="text">
              <string>Z:</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="rotate_left_x">
           <property name="text">
            <string>&lt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="rotate_right_x">
           <property name="text">
            <string>&gt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="rotate_left_y">
           <property name="text">
            <string>&lt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="rotate_right_y">
           <property name="text">
            <string>&gt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="rotate_left_z">
           <property name="text">
            <string>&lt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QPushButton" name="rotate_right_z">
           <property name="text">
            <string>&gt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="rot_x">
         <property name="font">
          <font>
           <family>Heiti SC</family>
          </font>
         </property>
         <property name="text">
          <string>0.00</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="rot_y">
         <property name="font">
          <font>
           <family>Heiti SC</family>
          </font>
         </property>
         <property name="text">
          <string>0.00</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="rot_z">
         <property name="font">
          <font>
           <family>Heiti SC</family>
          </font>
         </property>
         <property name="text">
          <string>0.00</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_24">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>scale</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="1" column="1">
        <widget class="QLabel" name="sc_y">
         <property name="font">
          <font>
           <family>Heiti SC</family>
          </font>
         </property>
         <property name="text">
          <string>0.00</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="sc_x">
         <property name="font">
          <font>
           <family>Heiti SC</family>
          </font>
         </property>
         <property name="text">
          <string>0.00</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" rowspan="4">
        <layout class="QGridLayout" name="gridLayout_6">
         <item row="0" column="0" rowspan="3">
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QLabel" name="label_18">
             <property name="font">
              <font>
               <family>Heiti SC</family>
              </font>
             </property>
             <property name="text">
              <string>X:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_19">
             <property name="font">
              <font>
               <family>Heiti SC</family>
              </font>
             </property>
             <property name="text">
              <string>Y:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_20">
             <property name="font">
              <font>
               <family>Heiti SC</family>
              </font>
             </property>
             <property name="text">
              <string>Z:</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="scale_left_y">
           <property name="text">
            <string>&lt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="scale_right_x">
           <property name="text">
            <string>&gt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="scale_right_y">
           <property name="text">
            <string>&gt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="scale_left_z">
           <property name="text">
            <string>&lt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="scale_left_x">
           <property name="text">
            <string>&lt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QPushButton" name="scale_left_all">
           <property name="text">
            <string>&lt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QPushButton" name="scale_right_z">
           <property name="text">
            <string>&gt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QPushButton" name="scale_right_all">
           <property name="text">
            <string>&gt;</string>
           </property>
           <property name="autoRepeat">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_22">
           <property name="text">
            <string>all:</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="sc_z">
         <property name="font">
          <font>
           <family>Heiti SC</family>
          </font>
         </property>
         <property name="text">
          <string>0.00</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLabel" name="sc_all">
         <property name="font">
          <font>
           <family>Heiti SC</family>
          </font>
         </property>
         <property name="text">
          <string>0.00</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>40</y>
      <width>231</width>
      <height>461</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_8">
     <item row="1" column="3" colspan="2">
      <widget class="QComboBox" name="type_of_protection">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <item>
        <property name="text">
         <string>central</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>parallel</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="4" column="0" colspan="2">
      <widget class="QLabel" name="label_29">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>color:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_28">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>type:</string>
       </property>
      </widget>
     </item>
     <item row="9" column="4">
      <widget class="QComboBox" name="size_of_verticles">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <item>
        <property name="text">
         <string>5</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>10</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>15</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="11" column="0" colspan="2">
      <widget class="QLabel" name="label_36">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>color:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="3" colspan="2">
      <widget class="QComboBox" name="type_of_edges">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <item>
        <property name="text">
         <string>solid</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>dotted</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="7" column="4">
      <widget class="QComboBox" name="display_of_verticles">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <item>
        <property name="text">
         <string>none</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>circle</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>square</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="5" column="3" colspan="2">
      <widget class="QComboBox" name="thickness_of_edges">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <item>
        <property name="text">
         <string>1.0</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>2.5</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>5</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="10" column="1" colspan="4">
      <widget class="QLabel" name="label_35">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>set background color</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2" colspan="3">
      <widget class="QLabel" name="label_27">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>settings of the edges</string>
       </property>
      </widget>
     </item>
     <item row="9" column="0">
      <widget class="QLabel" name="label_34">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>size:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0" colspan="3">
      <widget class="QLabel" name="label_30">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>thickness:</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1" colspan="4">
      <widget class="QLabel" name="label_31">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>settings of the verticles</string>
       </property>
      </widget>
     </item>
     <item row="7" column="0" colspan="4">
      <widget class="QLabel" name="label_32">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>display method:</string>
       </property>
      </widget>
     </item>
     <item row="8" column="0" colspan="2">
      <widget class="QLabel" name="label_33">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>color:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="4">
      <widget class="QPushButton" name="color_edges">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>choose </string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_26">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>type:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2" colspan="3">
      <widget class="QLabel" name="label_25">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>type of protection</string>
       </property>
      </widget>
     </item>
     <item row="8" column="4">
      <widget class="QPushButton" name="color_verticles">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>choose </string>
       </property>
      </widget>
     </item>
     <item row="11" column="4">
      <widget class="QPushButton" name="color_back">
       <property name="font">
        <font>
         <family>Heiti SC</family>
        </font>
       </property>
       <property name="text">
        <string>choose </string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="view_widget" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>20</y>
      <width>621</width>
      <height>591</height>
     </rect>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>view_widget</class>
   <extends>QOpenGLWidget</extends>
   <header>view_widget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
